BPMobileMessaging/Classes/ConactCenterCommunication/ContactCenterCommunicating.swift

```diff
@@ -4,74 +4,79 @@
 
 import Foundation
 
+/// Delegate an application should implement in order to receive events from the server.
 /// - Tag: ContactCenterEventsDelegating
 public protocol ContactCenterEventsDelegating: class {
+    /// Called when new events received from the server
+    /// - Parameter result: Array of `ContactCenterEvent` events
     func chatSessionEvents(result: Result<[ContactCenterEvent], Error>)
 }
 
 /// Provides chat and voice interactions.
 /// This API can be used for development of rich contact applications, such as customer-facing mobile and web applications for advanced chat, voice, and video communications with Bright Pattern Contact Center-based contact centers.
 /// Sends `poll` request to the backend repeatedly for get new chat events. The chat events are received through `delegate`
 public protocol ContactCenterCommunicating {
-    // MARK: - Client events delegate
-    /// Chat events delegate
+    // MARK: - Event delivery delegate
+    /// Chat events delegate.
     /// If successful returns an array of chat events [ContactCenterEvent](x-source-tag://ContactCenterEvent) for the current session that came from the server or [ContactCenterError](x-source-tag://ContactCenterError) otherwise
     /// - Tag: DelegateContactCenter
     var delegate: ContactCenterEventsDelegating? { get set }
-    // MARK:- Chat
-    /// Checks the current status of configured services
+    // MARK: - Requesting chat availability
+    /// Checks the current status of configured services.
     /// - Parameters:
     ///   - completion: Current status [ContactCenterServiceAvailability](x-source-tag://ContactCenterServiceAvailability) of configured services if successful or [ContactCenterError](x-source-tag://ContactCenterError) otherwise
     /// - Tag: checkAvailability
     func checkAvailability(with completion: @escaping ((Result<ContactCenterServiceAvailability, Error>) -> Void))
+    // MARK: - Requesting a new chat session
+    /// Request Chat initiates a chat session. It provides values of all or some of the expected parameters, and it may also contain the phone number of the mobile device. Note that if the mobile scenario entry is not configured for automatic callback, the agent can still use this number to call the mobile user manually, either upon the agent's own initiative or when asked to do this via a chat message from the mobile user.
```

Comment:
```suggestion
    /// Request Chat initiates a chat session. It provides values of all or some of the expected parameters, 
    /// and it may also contain the phone number of the mobile device. Note that if the mobile scenario entry is not
    /// configured for automatic callback, the agent can still use this number to call the mobile user
    /// manually, either upon the agent's own initiative or when asked to do this via a chat message from the mobile user.
```